lib/optimizer/src/TransformationEngine.php

```diff
@@ -112,7 +112,19 @@ private function getTransformerDependencies($transformerClass)
 
         $dependencies = [];
         foreach ($constructor->getParameters() as $parameter) {
-            $dependencyType = $parameter->getClass();
+            $dependencyType = null;
+
+            // The use of `ReflectionParameter::getClass()` is deprecated in PHP 8, and is superseded
+            // by `ReflectionParameter::getType()`. See https://github.com/php/php-src/pull/5209.
+            if (version_compare('7.1', PHP_VERSION, '<')) {
```

Comment:
No need for a complex `version_compare()` here, this can be done faster like this:
```suggestion
            if (PHP_VERSION_ID >= 70100)) {
```